### PR TITLE
Upgrade grpc to 150 and C++ to 14 For Cocoapods and SPM

### PR DESCRIPTION
**Pull request by wu-hui**

### CODE REVIEW

File: Package.swift [hunk 1]
```diff
@@ -175,12 +175,12 @@ let package = Package(
...
       name: "abseil",
       url: "https://github.com/firebase/abseil-cpp-SwiftPM.git",
-      "0.20220203.1" ..< "0.20220204.0"
+      "0.20220623.0" ..< "0.20220624.0"
     ),
     .package(
       name: "gRPC",
       url: "https://github.com/grpc/grpc-ios.git",
-      "1.44.0-grpc" ..< "1.45.0-grpc"
+      "1.50.0-grpc" ..< "1.50.1-grpc"
```
Comment (paulb777) [inline Package.swift:183]: To allow patches like the CocoaPods spec, this should be 1.51.0-grpc

**Comment by HannahShiSFB** [comment]
We have released 1.50.1, can you try if it passes

**Comment by wu-hui** [comment]
> We have released 1.50.1, can you try if it passes

Thanks, I am seeing build error however: https://github.com/firebase/firebase-ios-sdk/actions/runs/4599998333/jobs/8126084582?pr=10650

@sampajano FYI

**Comment by sampajano** [comment]
@HannahShiSFB Hi Hannah could you help take a look at the above error and if they ring a bell? Thanks!

**Review by cherylEnkidu — APPROVED** [review]

**Comment by wu-hui** [comment]
> @wu-hui, FYI ICYMI, #11208 updated this for SPM

Thank you!

**Pull request merged**